database/skylink.go

```diff
@@ -65,10 +65,17 @@ func (db *DB) Skylink(ctx context.Context, skylink string) (*Skylink, error) {
 		opts := options.Update().SetUpsert(true)
 		var ur *mongo.UpdateResult
 		ur, err = db.staticSkylinks.UpdateOne(ctx, filter, upsert, opts)
+		if err != nil {
+			return nil, err
+		}
 		// The UpsertedID might be nil in case the skylink got added to the DB
-		// by another server in between the calls.
-		if err == nil && ur.UpsertedID != nil {
+		// by another server in between the calls. In that case we'll fetch the
+		// skylink record from the DB.
+		if ur.UpsertedID != nil {
```

Comment:
Seems like this could have been avoided by using `db.staticSkylinks.FindOneAndUpdate` with upsert?
I think that would also clean the code up. 

Otherwise we'd need to add a regression test which tests the race. This way all we need to do is make sure that it's upserting correctly which I assume is tested already.